components/ratgdo/__init__.py

```diff
@@ -84,11 +84,11 @@ async def to_code(config):
 
     cg.add_library(
         name="secplus",
-        repository="https://github.com/esphome-ratgdo/secplus",
-        version="f98c3220356c27717a25102c0b35815ebbd26ccc",
+        repository="https://github.com/esphome-ratgdo/secplus#f98c3220356c27717a25102c0b35815ebbd26ccc",
+        version=None,
     )
     cg.add_library(
         name="espsoftwareserial",
-        repository="https://github.com/esphome-ratgdo/espsoftwareserial",
-        version="2f408224633316b997f82339e5b2731b1e561060",
+        repository="https://github.com/ratgdo/espsoftwareserial#ad8fd1cfd953fc11ea2227005e898f9671e202a7",
+        version=None,
```

Comment:
Let's merge the changes to `main` in the fork and use that here so we don't forget about them next time we update from upstream

Reply:
Sounds good to me to keep the pin but also merge to main

Reply:
I merged to main, but that see it breaks the CI build:

`espsoftwareserial/src/circular_queue/circular_queue.h:69:85: note: 'std::make_unique' is only available from C++14 onwards`

It builds fine on my computer, maybe the CI build uses an older version of esphome/platformio?

Anyway, I should revert `main`, because this breaks all builds for older versions which ignore the hash when library dependencies are declared like this:
```
cg.add_library(
        name="espsoftwareserial",
        repository="https://github.com/esphome-ratgdo/espsoftwareserial",
        version="2f408224633316b997f82339e5b2731b1e561060",
    )
```
I discovered the hash should be like this to be used:
```
cg.add_library(
        name="espsoftwareserial",
        repository="https://github.com/esphome-ratgdo/espsoftwareserial#2f408224633316b997f82339e5b2731b1e561060",
        version=None,
    )
```

Reply:
I just upgraded the [base image](https://github.com/esphome/esphome/pull/5473) in esphome a few days ago so we still have people building with older images/compilers. It will be in 2023.11.0 so we can use the newer compiler features next month

Ideally we adjust the softwareserial changes to pass the CI until we know everyone is on the newer images

Reply:
I reverted main on `ratgdo/espsoftwareserial` to the old hash `2f408224633316b997f82339e5b2731b1e561060`.